### PR TITLE
default id for donor, corrected ids for other blocks

### PR DESCRIPTION
**Pull request by hsjoshi28**
Default id generated for donor
Corrected id's for: Tissue Section, Blocks, and Dataset

### CODE REVIEW

File: registrations.schema.json
Comment (hsjoshi28) [inline registrations.schema.json:1]: Autogenerated file

File: src/core/normalizer.js [hunk 1]
```diff
@@ -36,23 +36,30 @@ export function normalizeRegistration(data, ruiLocationsDir) {
...
         const ruiLocation = ensureRuiLocation(block, ruiLocationsDir);
-
+        //objectIndex, object, objectType, ...ancestors
+        ensureId(
+          donorId, 
+          donor, 
+          'Donor', 
+          provider, 
+          provider.defaults ? provider.defaults : ''
+        );
```
Comment (hsjoshi28) [inline src/core/normalizer.js:54]: defauld id generator for Donor

File: src/core/normalizer.js [hunk 2]
```diff
@@ -36,23 +36,30 @@ export function normalizeRegistration(data, ruiLocationsDir) {
...
         ensureDonorLabel(donor, block);
         ensureDescription(donor, ruiLocation, provider);
         ensureLink(donor, provider, provider.defaults ? provider.defaults : '');
 
         ensureId(
           blockId,
           block,
-          '',
+          'TissueBlock',
```
Comment (hsjoshi28) [inline src/core/normalizer.js:62]: Corrected Object type

File: src/core/normalizer.js [hunk 3]
```diff
@@ -73,6 +80,7 @@ export function normalizeRegistration(data, ruiLocationsDir) {
           ensureId(
             sectionId,
             section,
+            'TissueSection',
```
Comment (hsjoshi28) [inline src/core/normalizer.js:83]: Corrected Object type

File: src/core/normalizer.js [hunk 4]
```diff
@@ -95,6 +103,7 @@ export function normalizeRegistration(data, ruiLocationsDir) {
             ensureId(
               datasetId,
               dataset,
+              'Dataset',
```
Comment (hsjoshi28) [inline src/core/normalizer.js:106]: Corrected Object type

File: src/utils/data-schema.js
Comment (hsjoshi28) [inline src/utils/data-schema.js:1]: Removed '@' from id for Dataset, section, and samples.

File: examples/simple/registrations.yaml
Comment (hsjoshi28) [inline examples/simple/registrations.yaml:1]: removed id from donor level and corrected default id

**Review by bherr2 — APPROVED** [review]

**Pull request merged**